module/executiondatasync/optimistic_syncing/core.go

```diff
@@ -122,87 +148,106 @@ func NewCoreImpl(
-		persister:                persisterComponent,
-		executionResult:          executionResult,
-		header:                   header,
-		inmemRegisters:           inmemRegisters,
-		inmemEvents:              inmemEvents,
-		inmemCollections:         inmemCollections,
-		inmemTransactions:        inmemTransactions,
-		inmemResults:             inmemResults,
-		inmemTxResultErrMsgs:     inmemTxResultErrMsgs,
-	}, nil
+		log: coreLogger,
+		processor: &executionDataProcessor{
+			execDataRequester:             execDataRequester,
+			txResultErrMsgsRequester:      txResultErrMsgsRequester,
+			txResultErrMsgsRequestTimeout: txResultErrMsgsRequestTimeout,
+			indexer:                       indexerComponent,
+			persister:                     persisterComponent,
+			inmemRegisters:                inmemRegisters,
+			inmemEvents:                   inmemEvents,
+			inmemCollections:              inmemCollections,
+			inmemTransactions:             inmemTransactions,
+			inmemResults:                  inmemResults,
+			inmemTxResultErrMsgs:          inmemTxResultErrMsgs,
+		},
+		executionResult: executionResult,
+		header:          header,
+	}
 }
 
-// Download implements the Core.Download method.
-// It downloads execution data for the block and stores it in the execution data cache.
+// Download downloads execution data for the block
 // Expected errors:
 // - context.Canceled: if the provided context was canceled before completion
-//
-// No other errors are expected during normal operations
+// All other errors are unexpected and may indicate a bug or inconsistent state
 func (c *CoreImpl) Download(ctx context.Context) error {
-	blockID := c.executionResult.BlockID
-	height := c.header.Height
-
-	c.log.Debug().
-		Hex("block_id", logging.ID(blockID)).
-		Uint64("height", height).
-		Msg("downloading execution data")
-
-	executionData, err := c.execDataRequester.RequestExecutionData(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to request execution data: %w", err)
+	c.log.Debug().Msg("downloading execution data")
+
+	g, gCtx := errgroup.WithContext(ctx)
+
+	var executionData *execution_data.BlockExecutionData
+	g.Go(func() error {
+		var err error
+		executionData, err = c.processor.execDataRequester.RequestExecutionData(gCtx)
+		if err != nil {
+			return fmt.Errorf("failed to request execution data: %w", err)
+		}
+
+		return nil
+	})
+
+	var txResultErrMsgsData []flow.TransactionResultErrorMessage
+	g.Go(func() error {
+		timeoutCtx, cancel := context.WithTimeout(gCtx, c.processor.txResultErrMsgsRequestTimeout)
+		defer cancel()
+
+		var err error
+		txResultErrMsgsData, err = c.processor.txResultErrMsgsRequester.Request(timeoutCtx)
+		if err != nil {
+			return fmt.Errorf("failed to request transaction result error messages data: %w", err)
+		}
+		return nil
+	})
+
+	if err := g.Wait(); err != nil {
+		//TODO: For me executionData and txResultErrMsgsData are not equally important. Could we proceed indexing without `txResultErrMsgsData`?
```

Comment:
yes. if downloading tx error messages fails or times out, we can skip it. the API should handle missing error messages gracefully, and there is a way for operators to backfill the error messages to recover.

we can iterate on this later to improve.

can you add a comment with these details in the code?